### DISCUSSION TITLE
Providing P2P free proxy solution for Iranian people

### DISCUSSION THREAD

**Discussion by farnoodma** [WeAreMahsaAmini/Core#6] [Ideas]
**Context**: The Internet is the main communication platform in Iran, the regime cannot shut down the internet for so long since most of their own infrastructure relies on the internet. However, they restrict Iranian access to the global network and they intend to create an intranet for the whole country.

**State**: Right now, all social platforms (including Twitter and Instagram) and all messaging apps (including Telegram and Whatsapp) are filtered in Iran. People have a hard time with VPNs and they rarely get access to the outside world.
Alongside those who have their own VPN, data centers have access to the global network, the reason is that businesses need access to outside resources. This was also the case in [Bloody Aban](https://en.wikipedia.org/wiki/2019%E2%80%932020_Iranian_protests).

**Idea**: By providing a reliable P2P proxy platform, we might be able to allow Iranians to use free internet while they are protesting and rioting. Android is the most used mobile OS in Iran and giving APK to people should be pretty easy (As you may know, AppStore and PlayStore are all filtered in Iran).

**Comment by norsnap** [comment]
check this https://github.com/WeAreMahsaAmini/Core/discussions/62

**Comment by ghost** [comment]
If you distribute APKs through Telegram and other messaging apps, the cyber army will start making and distributing their own copycat APKs.

**Comment by amastaneh** [comment]
In 2014, I attempted to share this draft copy with few friends in an effort to get their feedback on a concept I'd had years before.
I'd appreciate it if you could take a look at this draft and give us a feedback on what you think.

https://docs.google.com/document/d/1HzhiABu251rVWrDcJKV-85OmL8VXMku7FWVDDmYKklA/

**Reply by MaximusDecimusAnonymous** [reply]
Your idea is brilliant! Although it's like some existing protocols we don't have such a thing yet AFAIK.
There is some offline protocol apple uses to FindMyMac (or FindMyDevice) working with Bluetooth even when it doesn't have any connection. It's possible for other developers to develop such a thing and even adopt it broadly by tech companies.
We have a lot of computers around us and there is a high chance for at least one of them to connect to the outside synchronously or asynchronously.

**Comment by tcf-admin** [comment]
Hi @farnoodma, all, 

We are currently embedding [NewNode](https://github.com/clostra/newnode/) into [forDem](https://github.com/tcfev/fordem).
This gives us WiFi Direct and BLE connectivity. 
Content produced while in P2P mode will be then synced through a DHT which itself is connected to a series of decentralized servers.

*in another effort we are working on an open firmware for routers that enables them to run and join Mesh networks.
